autoscale-workload-rabbit/src/main/java/com/hpe/caf/autoscale/workload/rabbit/RabbitWorkloadAnalyser.java

```diff
@@ -103,11 +103,14 @@ public ScalingAction analyseWorkload(final InstanceInfo instanceInfo)
                 counter = 0;
                 int workersNeeded = getWorkersNeeded(stats.getMessages(), profile.getBacklogGoal(), instanceInfo);
                 LOG.debug("Workers needed to meet backlog goal: {}", workersNeeded);
-                if ( workersNeeded > instanceInfo.getTotalInstances() ) {
-                    int scale = Math.min(MAX_SCALE, workersNeeded - instanceInfo.getTotalInstances());
+                if ( workersNeeded > instanceInfo.getTotalRunningAndStageInstances() ) {
+                    int scale = Math.min(MAX_SCALE, workersNeeded - instanceInfo.getTotalRunningAndStageInstances());
                     return getScalingAction(ScalingOperation.SCALE_UP, scale);
-                } else if ( workersNeeded < instanceInfo.getTotalInstances() ) {
-                    return getScalingAction(ScalingOperation.SCALE_DOWN, instanceInfo.getTotalInstances() - workersNeeded);
+                } else if ( workersNeeded < instanceInfo.getTotalRunningAndStageInstances() ) {
+                    if(workersNeeded == 0){
```

Comment:
What is the purpose of the introduction of this `if` check?  The previous code looks simpler and equivalent.

Reply:
Previously if we had 4 instances running and our publish/consumption from rabbit was 0 we returned that we needed the same number of workers that we currently have, this has been changed to say that we need 0 workers to allow for rescaling to zero when work is completed.

Reply:
Re-evaluate.  That doesn't make sense.